src/methods/product/__tests__/product.test.js

```diff
@@ -21,19 +21,48 @@ describe('product', () => {
 
   it('returns double', () => {
     const iter = product([1, 2], [3, 4]);
-    expect(iter.getSize()).toEqual(4);
+    expect(iter.size).toEqual(4);
     expect(Array.from(iter)).toEqual([[1, 3], [1, 4], [2, 3], [2, 4]]);
   });
 
   it('returns with repeat', () => {
     const iter = product([1, 2], [1, 2]);
-    expect(iter.getSize()).toEqual(4);
+    expect(iter.size).toEqual(4);
     expect(Array.from(iter)).toEqual([[1, 1], [1, 2], [2, 1], [2, 2]]);
   });
 
+  it('returns a 4 x 4 product', () => {
+    const iter = product([1, 2, 3, 4], [1, 2, 3, 4]);
+    expect(iter.size).toEqual(16);
+    expect(Array.from(iter)).toEqual([
+      [1, 1],
+      [1, 2],
+      [1, 3],
+      [1, 4],
+      [2, 1],
+      [2, 2],
+      [2, 3],
+      [2, 4],
+      [3, 1],
+      [3, 2],
+      [3, 3],
+      [3, 4],
+      [4, 1],
+      [4, 2],
+      [4, 3],
+      [4, 4],
+    ]);
+  });
+
+  it('works with different length inputs', () => {
+    const iter = product([1, 2], [3, 4, 5]);
+    expect(iter.size).toEqual(6);
+    expect(Array.from(iter)).toEqual([[1, 3], [1, 4], [1, 5], [2, 3], [2, 4], [2, 5]]);
+  });
```

Comment:
You have two assertions in one `it`.

I would do this instead:

```javascript
  describe('works with different length inputs', () => {
    const iter = product([1, 2], [3, 4, 5]);
    
    it('size property', () => {
      expect(iter.size).toEqual(6);
    })

    it('contains expected elements', () => {
      expect(Array.from(iter)).toEqual([[1, 3], [1, 4], [1, 5], [2, 3], [2, 4], [2, 5]]);
    })
  });
```

Reply:
I didn't create this pattern here. And I really don't think there's anything necessarily wrong with having two assertions in one `it`.

Reply:
> I really don't think there's anything necessarily wrong with having two assertions in one `it`.

When it both assertions fail, you can only see the first error and not the second.

Reply:
I feel confident saying that I'm not worried about that problem since I screwed both parts of those assertions up several times while doing this refactor, and it was only the smallest of inconveniences that I didn't get both errors at once.